internal/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -253,9 +270,6 @@ func (m regoMetadata) getOrCreate(name string) map[string]interface{} {
 }
 
 func (r *RegoPolicyInterpreter) updateMetadata(ops []*regoMetadataOperation) error {
-	r.mutex.Lock()
```

Comment:
If this is removed because it `updateMetadata` is only to be called when `dataAndModulesMutex` is held, can you add a `// dataAndModulesMutex must be held before calling this` comment?

Reply:
Done.